opendevin/core/config.py

```diff
@@ -363,63 +362,62 @@ def load_from_toml(config: AppConfig, toml_file: str = 'config.toml'):
     # if there was an exception or core is not in the toml, try to use the old-style toml
     if 'core' not in toml_config:
         # re-use the env loader to set the config from env-style vars
-        load_from_env(config, toml_config)
+        load_from_env(cfg, toml_config)
         return
 
     core_config = toml_config['core']
 
     try:
         # set llm config from the toml file
-        llm_config = config.llm
         if 'llm' in toml_config:
-            llm_config = LLMConfig(**toml_config['llm'])
+            cfg.llm = LLMConfig(**toml_config['llm'])
 
         # set agent config from the toml file
-        agent_config = config.agent
         if 'agent' in toml_config:
-            agent_config = AgentConfig(**toml_config['agent'])
+            cfg.agent = AgentConfig(**toml_config['agent'])
 
-        # update the config object with the new values
-        config = AppConfig(llm=llm_config, agent=agent_config, **core_config)
+        # set the core config from the toml file
+        for key, value in core_config.items():
+            setattr(cfg, key, value)
```

Comment:
hmm does this potentially circumvent/miss any specifics (existing or in the future) that the AppConfig might do? 🤔

Reply:
the local variable won't work here, but it functions well due to the Singleton.

Reply:
I wondered the same thing. I _think_ it currently works okay...

Reply:
Is it possible that if the config.toml does not exist and/or there's no "core" section or the section is empty, that the "for" loop will run into an exception because of None values?

Reply:
> hmm does this potentially circumvent/miss any specifics (existing or in the future) that the AppConfig might do? 

I wondered the same thing. I _think_ it currently works okay...

Reply:
Why doesn't the local variable work? Just curious, it has to work, I thought.

Reply:
Instead of saying “this makes it work”:
```
config = AppConfig(llm=llm_config, agent=agent_config, **core_config)
```

it’s more accurate to say “this makes it work”:

```
AppConfig(llm=llm_config, agent=agent_config, **core_config)
```

maybe we just need to remove "config =" and keep everything else as-is. 😅

Reply:
ahh, right. it is not returning a config at all, so it's all the same, doh! 🤣

Reply:
Ah, good point, of course you're right!

Reply:
```
        for key, value in core_config.items():
            setattr(cfg, key, value)
```

after a second thought, I think this may not be a good practice, `AppConfig(llm=llm_config, agent=agent_config, **core_config)` looks more intuitive.

Reply:
Yes, I like AppConfig(stuffs). I like also that it reminds us _why_ it works. 🤷